### PR TITLE
Mingw/Win cleanups

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/meson.build [hunk 1]
```diff
@@ -104,6 +104,7 @@ if host_system == 'windows'
     add_project_arguments(msvc_args, language: 'c')
   else
     conf_data.set10('TARGET_MINGW_W64', true)
+    add_project_arguments(['-D__USE_MINGW_ANSI_STDIO=1'], language: 'c')
```
Comment (ubitux) [inline libnopegl/meson.build:107]: don't you need this in `ngl-tools` as well?

File: libnopegl/src/pthread_compat.h [hunk 2]
```diff
@@ -73,7 +73,7 @@ static inline int pthread_create(pthread_t *thread, const pthread_attr_t *attr,
      * conditions.
      * See: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createthread#remarks
      */
-    thread->handle    = (HANDLE)_beginthreadex(NULL, stack_size, pthread_compat_worker, thread, 0, NULL);
+    thread->handle    = (HANDLE)_beginthreadex(NULL, (unsigned)stack_size, pthread_compat_worker, thread, 0, NULL);
```
Comment (ubitux) [inline libnopegl/src/pthread_compat.h:76]: Note: you probably want this in `nope.media` as well

File: configure.py [hunk 3]
```diff
@@ -727,6 +727,46 @@ def get_env(self):
...
+        f"""\
+        ngli_deactivate() {{
+            deactivate
+            unset NGL_DLL_DIRS
+            unset -f ngli_deactivate
+        }}
+
+        . {activate_path}
+        export NGL_DLL_DIRS="{cfg.bin_path}"
+        """
```
Comment (ubitux) [inline configure.py:760]: do we really want to set this `NGL_DLL_DIRS` thing on macos and linux?

File: configure.py [hunk 4]
```diff
@@ -728,7 +725,6 @@ def get_env(self):
 
 
 def _build_env_scripts(cfg):
-
```
Comment (ubitux) [inline configure.py]: wrong commit

Comment (mbouron) [reply configure.py]: Fixed.

File: configure.py (hunk 3 above)
Comment (mbouron) [reply configure.py:760]: NGL_DLL_DIRS is ignored on platforms other than Windows, so it doesn't hurt to set it unconditionally.

File: libnopegl/src/pthread_compat.h (hunk 2 above)
Comment (mbouron) [reply libnopegl/src/pthread_compat.h:76]: Indeed and I've updated the commit to add it to ngl-tools.

File: libnopegl/meson.build (hunk 1 above)
Comment (mbouron) [reply libnopegl/meson.build:107]: Done.

**Review by ubitux — APPROVED** [review]

**Pull request merged**